src/mame/sinclair/spec_snqk.cpp

```diff
@@ -205,6 +205,15 @@ SNAPSHOT_LOAD_MEMBER(spectrum_state::snapshot_cb)
 
 		setup_frz(&snapshot_data[0], snapshot_size);
 	}
+	else if (image.is_filetype("spg"))
+	{
+		if (snapshot_data[32] != 'S' && snapshot_data[33] != 'p' && snapshot_data[34] != 'e' && snapshot_data[35] != 'c'
+			&& snapshot_data[36] != 't' && snapshot_data[37] != 'r' && snapshot_data[38] != 'u' && snapshot_data[39] != 'm'
+			&& snapshot_data[40] != 'P' && snapshot_data[41] != 'r' &&snapshot_data[42] != 'o' && snapshot_data[43] != 'g' )
+			return std::make_pair(image_error::INVALIDIMAGE, "Invalid .SPG file header.\n");
```

Comment:
If you’re checking for a literal string, you should be using `||`, when the condition is negative, not `&&`.  This will check for *one* valid character, not the whole string.

Remember De Morgan’s theorem:
* !(X && Y) → !X || !Y
* !((X == a) && (Y ==b)) → !(X == a) || !(Y == b) → (X != a) || (Y != b)

Also, there should be no newline on the end of the error string here.